debian/changelog

```diff
@@ -1,12 +1,14 @@
-singularity-container (2.5.1) unstable; urgency=high
+singularity-container (2.5.1-1) unstable; urgency=high
```

Comment:
Does the revision (i.e. the `-1`) *need* to exist? I have done a build on Debian 8/9 and without putting one in it defaulted to a `-1` revision without. I think...  Pretty sure I didn't make any changes my first few build attempts.

IIRC, there was also a comment made ages ago that the revision didn't need to be put there as the package maintainer would be modifying it...

Reply:
At least in the default configuration, building a debian package breaks at the `dpkg-source` step inside `debuild`. Using this container

```
bootstrap: docker
from: debian:8

%environment
    LC_ALL=C

%post
    apt-get update
    apt-get -y dist-upgrade
    apt-get -y install devscripts build-essential lintian gdebi-core git
```

will give

```
Singularity deb8.simg:~/devel/extern/singularity> git status
On branch development-2.x
Your branch is up-to-date with 'origin/development-2.x'.
nothing to commit, working directory clean
Singularity deb8.simg:~/devel/extern/singularity> dpkg-source -b .
parsechangelog/debian: warning: singularity_clean/debian/changelog(l9): found start of entry where expected more change data or trailer
LINE: singularity-container (2.5.0) unstable; urgency=high
parsechangelog/debian: warning: singularity_clean/debian/changelog(l9): found eof where expected more change data or trailer
dpkg-source: error: can't build with source format '3.0 (quilt)': non-native package version does not contain a revision
```

This goes away when you add the revision and the maintainer.

```
Singularity deb8.simg:~/devel/extern/singularity> dpkg-source -b .
dpkg-source: info: using source format `3.0 (quilt)'
dpkg-source: info: building singularity-container using existing ./singularity-container_2.5.1.orig.tar.gz
dpkg-source: info: building singularity-container in singularity-container_2.5.1-1.debian.tar.xz
dpkg-source: info: building singularity-container in singularity-container_2.5.1-1.dsc
```

It all comes down to what source format is set in `debian/source/format`. `3.0 (quilt)` for non-native wants the revision, `3.0 (native)` does not.

I'm no expert in debian package building, so maybe this is a non-issue due to some convention as you mentioned.

Related PR: #1736

Reply:
I'm fine with this.  Either way the maintainer has to edit the file.